File: AutoRest/Generators/Java/Java.Tests/src/test/java/fixtures/bodyboolean/BoolTests.java
package fixtures.bodyboolean;

import com.fasterxml.jackson.core.JsonParseException;

import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

import fixtures.bodyboolean.implementation.AutoRestBoolTestServiceImpl;

public class BoolTests {
    private static AutoRestBoolTestService client;

    @BeforeClass
    public static void setup() {
        client = new AutoRestBoolTestServiceImpl("http://localhost:3000");
    }

    @Test
    public void getNull() throws Exception {
        Assert.assertNull(client.bools().getNull().getBody());
    }

    @Test
    public void getInvalid() throws Exception {
        try {
            client.bools().getInvalid();
            Assert.assertTrue(false);
        } catch (Exception exception) {
            // expected
            Assert.assertEquals(JsonParseException.class, exception.getClass());
        }
    }

    @Test
    public void getTrue() throws Exception {
        boolean result = client.bools().getTrue().getBody();
        Assert.assertTrue(result);
    }

    @Test
    public void getFalse() throws Exception {
        boolean result = client.bools().getFalse().getBody();
        Assert.assertFalse(result);
    }

    @Test
    public void putTrue() throws Exception {
        client.bools().putTrue(true);
    }

    @Test
    public void putFalse() throws Exception {
        client.bools().putFalse(false);
    }
}

File: src/generator/AutoRest.Python.Tests/AcceptanceTests/bool_tests.py
# --------------------------------------------------------------------------
#
# Copyright (c) Microsoft Corporation. All rights reserved.
#
# The MIT License (MIT)
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the ""Software""), to
# deal in the Software without restriction, including without limitation the
# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
# sell copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
# IN THE SOFTWARE.
#
# --------------------------------------------------------------------------

import unittest
import subprocess
import sys
import isodate
import tempfile
import json
from datetime import date, datetime, timedelta
import os
from os.path import dirname, pardir, join, realpath

cwd = dirname(realpath(__file__))
log_level = int(os.environ.get('PythonLogLevel', 30))

tests = realpath(join(cwd, pardir, "Expected", "AcceptanceTests"))
sys.path.append(join(tests, "BodyBoolean"))

from msrest.serialization import Deserializer
from msrest.exceptions import DeserializationError

from auto_rest_bool_test_service import AutoRestBoolTestService
from auto_rest_bool_test_service.models import ErrorException

class BoolTests(unittest.TestCase):

    def test_bool(self):
        client = AutoRestBoolTestService(base_url="http://localhost:3000")

        self.assertTrue(client.bool_model.get_true())
        self.assertFalse(client.bool_model.get_false())

        client.bool_model.get_null()
        client.bool_model.put_false(False)
        client.bool_model.put_true(True)

        with self.assertRaises(ErrorException):
            client.bool_model.put_true(False)

        with self.assertRaises(DeserializationError):
            client.bool_model.get_invalid()

if __name__ == '__main__':
    unittest.main()
